### PR TITLE
Add content-based filtering recommendation system

### PR DESCRIPTION
**Pull request by patdel0**
The system suggests songs by calculating the Euclidean distance between the mean musical characteristics of each MBTI personality type and individual tracks in the two datasets.

In simpler terms, if user A belongs to type X, and type X fancies tracks with high energy, they will be suggested tracks with high energy.

The algorithm doesn't yet consider the user's liked songs. Once we've successfully integrated the ML environment with the application, I'll begin exploring the inclusion of the user's liked songs in the recommendations.

The [data analysis](https://github.com/JamesRobertSutcliffe/personality-music-recommender/blob/46d6d48539f22712878085416b0063367f722484/ml/content_based_recommender.ipynb) seems to have confirmed some biases (see "Data exploration" section if interested) such as: 
- Extroverted MBTI types enjoy high energy/danceable songs
- Introverted types enjoy more calmer, lower energy tracks
- Similar personality types enjoy similar songs

Although far from conclusive and based only on a limited dataset, it seems that personality types might in fact influence song preferences.

### CODE REVIEW

**Comment by patdel0** [comment]
@JamesRobertSutcliffe This only works out of the box if the Spotify API uses the `track_id` in this format: `47UUNYT0lsDnX9EYHIBf3L`

Whenever you get the chance, could you please confirm the format of the id used in their API to retrieve specific songs?

**Comment by JamesRobertSutcliffe** [comment]
Hi Patrick, this is very interesting. It will be a great basis for the recommendations for our app. 

Example value from spotify docs: "11dFghVXANMlKmJXsNCbNl"

This looks to be a match to me!

**Review by JamesRobertSutcliffe — APPROVED** [review]
Looks like this will be a great basis for recommendations on our app. If we could build on this with liked songs from our users down the line even better!
